File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int index = 1, end = 0; 
        int count = 0; 
        int n = chars.size(); 
        while(end < n){
            char curr = chars[end]; 
            while(end < n && chars[end] == curr){
                end++;
                count++; 
            }
            string tmp = to_string(count); 
            if(count > 1){
                for(char& c : tmp) chars[index++] = c; 
            }
            //cout << index << ' ' << end << endl; 
            if(end < n) chars[index++] = chars[end]; 
            count = 0; 
            curr = chars[end]; 
        }

        return index; 
    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        int index = 0; 
        int end = 0; 
        while(end < chars.length){
            char curr = chars[end]; 
            int count = 0; 
            while(end < chars.length && chars[end] == curr){
                count++; 
                end++; 
            }

            chars[index++] = curr; 
            if(count > 1){
                String num = String.valueOf(count); 
                for(char c : num.toCharArray()) chars[index++] = c; 
            }
        }

        return index; 
    }
}